.editorconfig

```diff
@@ -106,6 +106,9 @@ dotnet_style_predefined_type_for_member_access = true:suggestion
 #prefer the language keyword for local variables, method parameters, and class members, instead of the type name, for types that have a keyword to represent them
 dotnet_style_predefined_type_for_locals_parameters_members = true:suggestion
 
+#don't require access modifiers, allowing omitting of default modifiers (`omit_if_default` doesn't seem to work?)
+dotnet_style_require_accessibility_modifiers = never
```

Comment:
Try `omit_if_default:suggestion` instead of `never`.

`never` = Doesn't complain when modifiers are present or missing. Basically just shuts it up.
`omit_if_default` = Complain to remove modifiers when they are the default. This is what we'd want.

The default severity appears to be `silent` - so you need to invoke fixers from your IDE on the violating line to see it. Bumping it to suggestion will get it to show in the IDE Info/Messages list.

Reply:
Is `suggestion` strong enough to trigger a style failure on the CI/`make check`?

Reply:
No, the `:<severity>` syntax is IDE only. So `omit_if_default:error` would show as an error in your IDE, but builds would not care.

For build-time checking you need to use the `dotnet_diagnostic.<rule>.severity = <severity>` syntax instead. Plus you need `EnforceCodeStyleInBuild` true for it to run.

Since `make check` is currently set with `-warnaserror -p:EnforceCodeStyleInBuild=true` we can cause it to fail by setting any style rule to a warning level. This would appear as a warning in the IDE and in regular builds, but as an error in `make check`.

So if we wanted, we could enforce this (or other rules) with `dotnet_diagnostic.IDE0040.severity = warning`

I would leave that as an exercise for another PR though.

Reply:
> Try omit_if_default:suggestion instead of never.
The idea with `never` here was to silence ReSharper, who would constantly nudge me to add an explicit modifier as that is C# default (but not OpenRA default).

Using `omit_if_default:warning` on the other hand raises a very misleading warning: "Accessibility modifiers required" when you add a modifier that you need to remove 🤦‍♂️ 

So this may be a dead end.

Reply:
Never mind I think I made it work, thanks.